### DISCUSSION TITLE
[Question] label text_align

### DISCUSSION THREAD

**Discussion by 3mora2** [flet-dev/flet#2439] [Q&A]
### Question

how change text_align of TextField label to right 
value can changed by `text_align=ft.TextAlign.RIGHT`

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by burhansvural** [comment]
`
<code>
from flet import *

def main(page: Page):
    
    page.window_width=500
    page.window_height=990

    page.window_top=30
    page.window_left=1400

    page.title="Textfield Value Alignment Example- version : 1.0.1"

    page.horizontal_alignment = CrossAxisAlignment.CENTER
    page.auto_scroll = True
    page.scroll = ScrollMode.HIDDEN
    page.appbar = AppBar(
        title=Text(
            "TextField Demo App", weight=FontWeight.W_600, color=colors.BLACK87
        ),
        bgcolor=colors.BLUE,
        center_title=True,
        actions=[
            IconButton(icons.MENU, tooltip="Menu", icon_color=colors.BLACK87)
        ],
        color=colors.WHITE,
    )    
    txt_example=TextField(
            value="Text Field Value",
            label_style=TextStyle(
                weight=FontWeight.BOLD,
                ),
            text_align=TextAlign.LEFT,
            color=colors.BLACK87
        )

    def fab_pressed(e):
        if txt_example.text_align==TextAlign.LEFT:
            txt_example.text_align=TextAlign.CENTER
            page.update()
        elif txt_example.text_align==TextAlign.CENTER:
            txt_example.text_align=TextAlign.RIGHT
            page.update()
        elif txt_example.text_align==TextAlign.RIGHT:
            txt_example.text_align=TextAlign.START
            page.update()
        elif txt_example.text_align==TextAlign.START:
            txt_example.text_align=TextAlign.END
            page.update()
        elif txt_example.text_align==TextAlign.END:
            txt_example.text_align=TextAlign.LEFT
            page.update()

        page.show_snack_bar(
            SnackBar(Text(f"Textfield value align changed. Align={txt_example.text_align}"), open=True),
        )
        

    page.floating_action_button = FloatingActionButton(
        icon=icons.CHANGE_CIRCLE,on_click=fab_pressed,
        bgcolor=colors.LIME_300,
        text="Change TextField Value Align"
    )
    
    page.add(txt_example)
    page.update()
   

app(target=main)
</code>
`

**Reply by 3mora2** [reply]
Thanks for your reply, 
when add label and error_text for TextField, only value thant align change not label nor error_text 
try use:
```python
    txt_example = TextField(
        value="Text Field Value",
        label="Fill Text Field",
        label_style=TextStyle(
            weight=FontWeight.BOLD,
        ),
        text_align=TextAlign.LEFT,
        color=colors.BLACK87,
        error_text="Some error"
    )

```

**Reply by 3mora2** [reply]
there same thing like [Directionality](https://itecnote.com/tecnote/flutter-textfield-with-labeltext-on-right/)

**Reply by burhansvural** [reply]
It is necessary to add more advanced properties for the TextField's label property, but these have not been added yet.

**Reply by 3mora2** [reply]
ok

**Reply by ndonkoHenri** [reply]
Can [`page.rtl`](https://flet.dev/docs/controls/page#rtl) be of help?

**Reply by 3mora2** [reply]
yes, it good for me
but it change direction of all element, in app there Arabic input and English
should only change Arabic input to right

**Reply by FeodorFitsner** [reply]
Created new issue: https://github.com/flet-dev/flet/issues/2462

**Comment by burhansvural** [comment]
I have shown how this can be done with an easy technique in the application below. It's up to you to develop further.

![CustomTextField](https://github.com/flet-dev/flet/assets/149883930/8a84a911-37a0-4755-9485-7672868c77c6)
`
from enum import Enum
from flet import *
import flet.canvas as fcvs


class LabelAlign(Enum):
    NONE = None
    START = "start"
    CENTER = "center"
    END = "end"


class CustomTextField(fcvs.Canvas):
    __pos = LabelAlign.NONE

    def __init__(self, lblAlign: LabelAlign, lblStr: str, lblColor: str):
        super().__init__()
        self.__txt_width_widget = TextField(focused_border_width=1, focused_border_color=lblColor,
                                            border_color=lblColor)
        if lblAlign == LabelAlign.END:
            self.__pos = alignment.top_right
        elif lblAlign == LabelAlign.CENTER:
            self.__pos = alignment.top_center
        else:
            self.__pos = alignment.top_left
        self.__lblColor = lblColor
        self.__lblStr = "  " + lblStr + "  —"

        self.content = Stack(
            controls=[
                Container(
                    content=self.__txt_width_widget,
                ),
                Container(
                    content=Text(
                        overflow=TextOverflow.ELLIPSIS,
                        color=self.__lblColor,
                        value=self.__lblStr,
                        size=12,
                        bgcolor=colors.WHITE,
                    ),
                    alignment=self.__pos,
                    margin=margin.only(left=16, top=-9.88, right=6),
                )
            ],
        )


def main(pagex: Page):
    pagex.title = "Custom Text Field With Label Alignment"

    pagex.add(CustomTextField(lblAlign=LabelAlign.START, lblStr="Example Label 1", lblColor=colors.RED))
    pagex.add(CustomTextField(lblAlign=LabelAlign.CENTER, lblStr="Example Label 2", lblColor=colors.GREEN))
    pagex.add(CustomTextField(lblAlign=LabelAlign.END, lblStr="Example Label 3", lblColor=colors.ORANGE))
    pagex.update()


app(target=main)

`
